editor/components/inserter-with-shortcuts/index.js

```diff
@@ -44,17 +44,14 @@ function InserterWithShortcuts( { items, isLocked, onInsert } ) {
 }
 
 export default compose(
-	withEditorSettings( ( settings ) => {
-		const { templateLock, allowedBlockTypes } = settings;
-
+	withSelect( ( select ) => {
+		const { getEditorSettings, getFrecentInserterItems } = select( 'core/editor' );
+		const { templateLock, allowedBlockTypes } = getEditorSettings();
 		return {
+			items: getFrecentInserterItems( allowedBlockTypes, 4 ),
```

Comment:
Since it's in state now, we could improve this selector to avoid the `allowedBlockTypes` argument.

Reply:
Nice 👍